openapi/openapi/src/main/scala/endpoints4s/openapi/JsonSchemas.scala

```diff
@@ -74,14 +134,78 @@ trait JsonSchemas extends algebra.JsonSchemas with TuplesSchemas {
         documentation: Option[String]
     )
 
-    case class DocumentedCoProd(
-        alternatives: List[(String, DocumentedRecord)],
-        name: Option[String] = None,
-        discriminatorName: String = defaultDiscriminatorName,
-        description: Option[String] = None,
-        example: Option[ujson.Value] = None,
-        title: Option[String] = None
-    ) extends DocumentedJsonSchema
+    sealed abstract class DocumentedCoProd extends DocumentedJsonSchema {
+      def alternatives: List[(String, DocumentedRecord)]
+      def name: Option[String]
+      def discriminatorName: String
+
+      def withName(name: String): DocumentedCoProd
+      def withExample(example: => ujson.Value): DocumentedCoProd
+      def withTitle(title: String): DocumentedCoProd
+      def withDescription(description: String): DocumentedCoProd
+      def withDiscriminatorName(discriminatorName: String): DocumentedCoProd
+    }
+
+    object DocumentedCoProd {
+      case class Immediate(
+          alternatives: List[(String, DocumentedRecord)],
+          name: Option[String] = None,
+          discriminatorName: String = defaultDiscriminatorName,
+          description: Option[String] = None,
+          example: Option[ujson.Value] = None,
+          title: Option[String] = None
+      ) extends DocumentedCoProd {
+        def withName(name: String): DocumentedCoProd =
+          copy(name = Some(name))
+        def withExample(example: => ujson.Value): DocumentedCoProd =
+          copy(example = Some(example))
+        def withTitle(title: String): DocumentedCoProd =
+          copy(title = Some(title))
+        def withDescription(description: String): DocumentedCoProd =
+          copy(description = Some(description))
+        def withDiscriminatorName(discriminatorName: String): DocumentedCoProd =
+          copy(discriminatorName = discriminatorName)
+      }
+
+      class Lazy(n: String, docs: => DocumentedCoProd) extends DocumentedCoProd {
+        lazy val evaluatedDocs = docs
+        def alternatives: List[(String, DocumentedRecord)] = evaluatedDocs.alternatives
+        def name: Option[String] = Some(n)
+        def discriminatorName: String = evaluatedDocs.discriminatorName
+        def description: Option[String] = evaluatedDocs.description
+        def example: Option[ujson.Value] = evaluatedDocs.example
+        def title: Option[String] = evaluatedDocs.title
+
+        def withName(n: String): DocumentedCoProd =
+          new Lazy(n, docs)
+        def withExample(e: => ujson.Value): DocumentedCoProd =
+          new Lazy(n, this) {
```

Comment:
I think we should do the same everywhere, right?
```suggestion
          new Lazy(n, docs) {
```

Reply:
I don't think so because if we chain multiple `.withExample(...).withTitle(...)` the instance returned by `withTitle` would not return the example set by `withExample`.